core/src/parking_lot.rs

```diff
@@ -440,9 +473,11 @@ unsafe fn lock_bucket_pair<'a>(key1: usize, key2: usize) -> (&'a Bucket, &'a Buc
 // Unlock a pair of buckets
 #[inline]
 unsafe fn unlock_bucket_pair(bucket1: &Bucket, bucket2: &Bucket) {
+    // REVIEW: perhaps `ptr::eq`?
     if bucket1 as *const _ == bucket2 as *const _ {
         bucket1.mutex.unlock();
     } else if bucket1 as *const _ < bucket2 as *const _ {
+        // REVIEW: why does the order of unlocking matter here?
```

Comment:
Good point, it doesn't actually matter here.

Reply:
I'll simplify `unlock_bucket_pair`